src/ray/core_worker/core_worker.cc

```diff
@@ -528,10 +545,18 @@ Status CoreWorker::SubmitActorTask(const ActorID &actor_id, const RayFunction &f
 
   // Submit task.
   Status status;
-  if (is_direct_call) {
-    status = direct_actor_submitter_->SubmitTask(builder.Build());
+  const auto spec = builder.Build();
+  auto state = actor_handle->ActorState();
+  if (state.has_value() && *state == gcs::ActorTableData::DEAD) {
```

Comment:
if (!actor_handle->IsAlive())

Reply:
We can't just check `IsAlive` because that might mean the actor is restarting.